funnel/views/participant.py

```diff
@@ -144,18 +159,66 @@ def participant_badge(profile, space, participant):
     (Profile, {'name': 'profile'}, 'g.profile'),
     ((ProposalSpace, ProposalSpaceRedirect), {'name': 'space', 'profile': 'profile'}, 'space'),
     (Event, {'name': 'name', 'proposal_space': 'space'}, 'event'),
-    (Participant, {'id': 'participant_id'}, 'participant'),
     permission='checkin-event')
-def event_checkin(profile, space, event, participant):
-    attendee = Attendee.get(event, participant)
+def event_checkin(profile, space, event):
     form = forms.Form()
     if form.validate_on_submit():
-        # Toggle check-in status
-        attendee.checked_in = not attendee.checked_in
+        checked_in = True if request.form.get('checkin') == 't' else False
+        participant_ids = request.form.getlist('pid')
```

Comment:
How many ids does the typical request include? Doing two database queries per participant id seems grossly inefficient.

Reply:
Changed API to `'/<space>/event/<name>/participants/checkin'`
This API is meant for batch checkin of participants.
@jace Update it to `attendee = Attendee.get(event, participant_id)`, would that be fine?